engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/immutable/FromArrayBuiltinVectorNode.java

```diff
@@ -21,29 +18,19 @@ static FromArrayBuiltinVectorNode build() {
     return FromArrayBuiltinVectorNodeGen.create();
   }
 
-  abstract Vector execute(Object arr);
+  abstract EnsoObject execute(Object arr);
 
-  @Specialization
-  Vector fromVector(Vector arr) {
-    return arr;
-  }
-
-  @Specialization(guards = "interop.hasArrayElements(arr)")
-  Vector fromArrayLikeObject(
-      Object arr,
-      @Cached CoerceArrayNode coerce,
-      @CachedLibrary(limit = "3") InteropLibrary interop) {
-    return Vector.fromArray(new Array(coerce.execute(arr)));
+  @Specialization(guards = "isImmutableArrayLike(arr)")
+  EnsoObject fromVector(Object arr) {
+    return (EnsoObject) arr;
   }
 
-  @Fallback
-  Vector fromUnknown(Object arr) {
-    throw unsupportedException(arr);
+  @Specialization
```

Comment:
Add `guards = "!isImmutableArrayLike(arr)"` to the spec guard? Have you forgotten about it, or is there no guard in this spec on purpose?

Reply:
You are right. I tend to think of `@Specialization`s as a sequence (at least in this PR) without considering they may be applied in a random order.

It is not that huge problem (e.g. nothing is going to crash) when dealing with _array-like_ objects - both `Vector` and `Array` support `InteropLibrary.hasArrayElements` and if they mistakenly enter generic specialization, they will behave fine being treated in a generic way.

Reply:
In this particular case I am going to use `@Fallback` annotation.